dotnet/src/Connectors/Connectors.AzureAIInference/Extensions/AzureAIInferenceKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Azure.AI.Inference;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.AzureAIInference;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Sponsor extensions class for <see cref="IKernelBuilder"/>.
+/// </summary>
+public static class AzureAIInferenceKernelBuilderExtensions
+{
+    /// <summary>
+    /// Adds the Azure AI Inference chat completion service to the list.
+    /// </summary>
+    /// <param name="builder">The <see cref="IKernelBuilder"/> instance to augment.</param>
+    /// <param name="modelId">Target Model Id for endpoints supporting more than one model</param>
+    /// <param name="apiKey">API Key</param>
+    /// <param name="endpoint">Endpoint / Target URI</param>
+    /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <returns>The same instance as <paramref name="builder"/>.</returns>
+    public static IKernelBuilder AddAzureAIInferenceChatCompletion(
+        this IKernelBuilder builder,
+        string? modelId = null,
+        string? apiKey = null,
+        Uri? endpoint = null,
+        string? serviceId = null,
+        HttpClient? httpClient = null)
```

Comment:
Since these methods are new, is this a right moment to introduce `Options` class for such methods, at least for this connector? If we know that we are going to do that for other connectors, it makes sense to do it here now, rather than add it now with optional parameter and still update it later.

Reply:
@dmytrostruk Thanks for the suggestion, that is currently captured in here
- #6878

Will work in a ADR for this, and once it's agreed can start changing the connectors, we definitely will need a decision later on this one, as my first attempt when implementing in the Anthropic connectors was clearly something that needed more discussion.